cpu/esp_common/periph/i2c_sw.c

```diff
@@ -573,6 +595,7 @@ static IRAM_ATTR int _i2c_start_cond(_i2c_bus_t* bus)
         /* SCL = passive HIGH (floating and pulled-up) */
         _i2c_scl_high(bus);
 
+#if I2C_CLOCK_STRETCH > 0
         /* clock stretching, wait as long as clock is driven to low by the slave */
         uint32_t stretch = I2C_CLOCK_STRETCH;
         while (stretch && !_i2c_scl_read(bus)) {
```

Comment:
Are those `#ifdefs` really needed?
The compieler should be able to optimize the

```C
while (0 && !_i2c_scl_read(bus)) {
```

block away in that case anyway.

Reply:
I didn't have as much hope as you do considering that this is the espressif chip, but hey I checked it and indeed the compiler throws all that away without the ifdefs